slack/context.w

```diff
@@ -0,0 +1,145 @@
+bring "./message.w" as msg;
+bring "./util.w" as helpers;
+bring "./events.w" as events;
+
+/// The bahvioral interface of a thread
+pub inflight interface IThread {
+  inflight postMessage(message: msg.Message): Json;
+  inflight post(message: str): Json;
+}
+
+/// Represents the context of a slack channel
+pub inflight class Channel impl IThread {
+  /// The channel id
+  pub id: str;
+  botToken: str;
+  
+  new (channel: str, botToken: str) {
+    this.id = channel;
+    this.botToken = botToken;
+  }
+
+  /// Post a message block to a channel
+  pub inflight postMessage(message: msg.Message): Json {
+    return helpers.SlackUtils.post(
+      {
+        channel: this.id,
+        blocks: message.toJson(),
+      },
+      this.botToken
+    );
+  }
+
+  /// Post raw text to a channel
+  pub inflight post(message: str): Json {
+    return helpers.SlackUtils.post(
+      {
+        channel: this.id,
+        text: message
+      },
+      this.botToken
+    );
+  }
+}
+
+/// Represents the context of a slack thread
+pub inflight class Thread impl IThread {
+  /// The channel context
+  pub channel: Channel;
+  /// The thread timestamp
+  pub timestamp: str;
+  botToken: str;
+
+  new(channel: str, thread_ts: str, botToken: str) {
+    this.channel = new Channel(channel, botToken);
+    this.timestamp = thread_ts;
+    this.botToken = botToken;
+  }
+
+  /// Post a message to a thread
+  pub inflight postMessage(message: msg.Message): Json {
+    return helpers.SlackUtils.post(
+      {
+        channel: this.channel.id,
+        thread_ts: this.timestamp,
+        blocks: message.toJson()
+      },
+      this.botToken
+    );
+  } 
+
+  /// Post raw text to a thread
+  pub inflight post(message: str): Json {
+    return helpers.SlackUtils.post(
+      {
+        channel: this.channel.id,
+        thread_ts: this.timestamp,
+        text: message
+      },
+      this.botToken
+    );
+  }
+}
+
+pub inflight class Channel_Mock extends Channel {
```

Comment:
Why is this public?

Why the underscore in the name?

I haven't seen any documentation about this.

Reply:
Yea I can probably refactor this to make it not public.

I took the underscore convention from how the sim mocks in `bedrock` were defined.